File: 36/go/36.go
import (
    "fmt"
)

func isValidSudoku(board [][]byte) bool {
    // use map to record repeatition
    n := len(board)
    rowCheck := make([]map[byte]int, 9)
    colCheck := make([]map[byte]int, 9)
    boxCheck := make([]map[byte]int, 9)

    for i := 0; i< n; i++ {
        rowCheck[i] = make(map[byte]int)
        colCheck[i] = make(map[byte]int)
        boxCheck[i] = make(map[byte]int)
    }

    for i := 0; i < n; i++ {
        for j := 0; j < n; j++{
            str := board[i][j]
            if str == '.'{
                continue
            }

            // map the rows and columns
            r := i
            c := j
            // map the nine boxes
            b := (i/3)*3 + (j/3)

            if rowCheck[r][str] > 0 || colCheck[c][str] > 0 || boxCheck[b][str] > 0{
                return false
            }

            rowCheck[r][str] += 1
            colCheck[c][str] += 1
            boxCheck[b][str] += 1

        }
    }
    return true
}

File: 36/py/36.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        # make lists for checking
        rowCheck = [[0 for _ in range(9)] for _ in range(9)]
        colCheck = [[0 for _ in range(9)] for _ in range(9)]
        boxCheck = [[0 for _ in range(9)] for _ in range(9)]

        n = len(board)

        for i in range(n) :
            for j in range(n) :
                chr = board[i][j]
                num = 0
                if chr == '.' :
                    continue
                else :
                    num = int(chr) - 1
                
                # defind the id
                r = i
                c = j
                b = (i//3)*3 + (j//3)

                if rowCheck[r][num] > 0 or colCheck[c][num] > 0 or boxCheck[b][num] > 0 :
                    return False

                rowCheck[r][num] += 1
                colCheck[c][num] += 1
                boxCheck[b][num] += 1

        return True
